server/src/app.controller.ts

```diff
@@ -1,57 +1,52 @@
-import { Controller, Get, Post, Body } from '@nestjs/common';
-import { TestService, BrowseService } from './app.service';
+import { Body, Controller, Get, Post } from '@nestjs/common';
 import ApiFactory from './api/import';
-
+import * as Dtos from './app.dtos';
```

Comment:
I wanted to abstract the Dtos out of controller and just let various services keep track of them. Otherwise we're gonna have to make multiple points of edit if we wanted to change something to a new API. 

That's why I think not giving body a type, and letting the return be an arbitrary promise type is better. We can just disable eslint if it's unhappy. 

Thoughts?

Reply:
It shouldn't add any more points of edit than we already need. I for sure need the dtos somewhere; controller can just use the same interfaces as service functions. 

If anything the dtos would be removed from service functions into controller. The dtos are strongly related with the controller functions (every endpoint has exactly two corresponding dtos). The service functions don't have to follow this pattern.